### PR TITLE
syslog: Add sc_write_force callback

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- syslog: Remove syslog_force and related stuff 
- syslog: Add sc_write_force callback 

## Impact

improve the syslog speed in the interrupt context

### CODE REVIEW

File: drivers/syslog/syslog_flush.c [hunk 1]
```diff
@@ -64,6 +64,7 @@
 
 int syslog_flush(void)
 {
+  FAR struct syslog_channel_s *channel;
```
Comment (pkarashchenko) [inline drivers/syslog/syslog_flush.c:67]:
```suggestion
```

File: drivers/syslog/syslog_flush.c [hunk 2]
```diff
@@ -76,16 +77,17 @@ int syslog_flush(void)
 
   for (i = 0; i < CONFIG_SYSLOG_MAX_CHANNELS; i++)
     {
-      if (g_syslog_channel[i] == NULL)
+      channel = g_syslog_channel[i];
```
Comment (pkarashchenko) [inline drivers/syslog/syslog_flush.c:80]:
```suggestion
      FAR struct syslog_channel_s *channel = g_syslog_channel[i];
```

**Review by pkarashchenko — APPROVED** [review]

File: drivers/syslog/syslog_flush.c (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/syslog/syslog_flush.c]: Done.

File: drivers/syslog/syslog_flush.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/syslog/syslog_flush.c]: Done.

**Review by acassis — APPROVED** [review]

File: drivers/syslog/syslog_channel.c [hunk 3]
```diff
@@ -91,6 +91,7 @@ static const struct syslog_channel_ops_s g_rpmsg_channel_ops =
   syslog_rpmsg_putc,
   syslog_rpmsg_putc,
   syslog_rpmsg_flush,
+  syslog_rpmsg_write,
```
Comment (TaiJuWu) [inline drivers/syslog/syslog_channel.c:94]: It is same with below line.

**Pull request merged**